grr/server/grr_response_server/gui/ui/components/client/client.ts

```diff
@@ -46,6 +51,36 @@ export class Client implements OnInit, OnDestroy {
         });
   }
 
+  ngAfterViewInit() {
+    this.clientDetailsDrawers.closedStart.subscribe(() => {
+      const urlTokens = this.location.path().split('/');
+      this.location.go(urlTokens.slice(0, -1).join('/'));
+    });
+
+    this.location.onUrlChange(url => {
+      const urlTokens = url.split('/');
+      if (urlTokens[urlTokens.length - 1] === 'details') {
+        this.clientDetailsDrawers.open();
+      }
+    });
+
+    this.route.url.pipe(map(url => url[url.length - 1]), take(1))
```

Comment:
There is no need to take more than the first emitted value. We want to see the URL when the component loads, in order to know whether or not to open the drawer. After that, there is no need to still receive updates, so we can remove the subscription instance.

Reply:
After that, there **is** a need to receive updates, but this is handled by your `this.location.onUrlChange` code. Manipulating the URLs both via Router and Location seems like a code smell to me. Ideally, we would use Router/ActivatedRoute for both. Possibly, registering `/details` as a child route would allow us to keep the `Client` page loaded and load the drawer without needing to manipulate URLs so much ourselves.

This refactoring is not blocking for this PR, but please look into it, if you find the time.

Reply:
I agree, having only one function to handle the URL/Route change is preferred. I tried using only Router at first, but it reloaded the page, which wasn't what we wanted. I'll look into child routes, see if it works that way. I'll address this issue in #836.